src/core/main.js

```diff
@@ -462,9 +462,10 @@ for (const k in constants) {
  * ```
  *
  * Code placed in `setup()` will run once before code placed in
- * <a href="#/p5/draw">draw()</a> begins looping. If the
- * <a href="#/p5/preload">preload()</a> is declared, then `setup()` will
- * run immediately after <a href="#/p5/preload">preload()</a> finishes
+ * <a href="#/p5/draw">draw()</a> begins looping. When `setup()` is declared async, 
+ * execution pauses at each `await` until the promise resolves, ensuring all assets 
```

Comment:
Similarly, providing a bit of explanation on promise:

"execution pauses at each `await`: for example, `font = await loadFont(...)` will wait for the font asset to load. This is because the `loadFont(..)` function returns a promise, and the `await` keyword means the program will wait for the promise to resolve."

Feel free to rephrase of course.